core/templates/dev/head/galleries/Gallery.js

```diff
@@ -281,21 +251,34 @@ oppia.controller('Gallery', [
 oppia.controller('SearchBar', [
   '$scope', '$rootScope', '$translate', 'searchService', 'oppiaDebouncer',
   'ExplorationCreationButtonService', 'urlService', 'CATEGORY_LIST',
+  'i18nIdService',
   function(
       $scope, $rootScope, $translate, searchService, oppiaDebouncer,
-      ExplorationCreationButtonService, urlService, CATEGORY_LIST) {
+      ExplorationCreationButtonService, urlService, CATEGORY_LIST,
+      i18nIdService) {
+    $rootScope.$on('$translateChangeSuccess', function() {
+      // This strings must be translated with a filter, generating FoUC for
```

Comment:
It's horribly written. It should be "If you translate this strings in the html, then you must use a filter because only the first 14 characters are displayed. That would generate FOUC for languages other than English. As an exception, we translate them here and update the translation every time the language is changed." Is it better?